pkg/csi/node.go

```diff
@@ -367,6 +363,41 @@ func (ns *nodeService) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequ
 
 }
 
+func (ns *nodeService) NodeGetVolumeStats(ctx context.Context,
+	req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
+
+	klog.Infof("NodeGetVolumeStats called with req: %#v", req)
+
+	volumePath := req.GetVolumePath()
+	if volumePath == "" {
+		klog.Errorf("unable to get volume path from request")
+		return nil, fmt.Errorf("unable to get volume path from request")
+	}
+
+	var statFS unix.Statfs_t
+	if err := unix.Statfs(volumePath, &statFS); err != nil {
+		klog.Errorf("unable to get stats of volume [%s]: [%v]", volumePath, err)
+		return nil, fmt.Errorf("unable to get stats of volume [%s]: [%v]", volumePath, err)
+	}
+
+	return &csi.NodeGetVolumeStatsResponse{
+		Usage: []*csi.VolumeUsage{
+			{
+				Available: int64(statFS.Bavail) * int64(statFS.Bsize),
+				Total: int64(statFS.Blocks) * int64(statFS.Bsize),
+				Used: (int64(statFS.Blocks) - int64(statFS.Bavail)) * int64(statFS.Bsize),
+				Unit: csi.VolumeUsage_BYTES,
+			},
+			{
+				Available: int64(statFS.Ffree),
```

Comment:
Just curious, Why are there exactly two elements in the array?

Reply:
CSI expects this.  The first shares the byte statistics and the second shares the index node (inode) statistics.  This is how the Kubernetes container storage interface expects a return.